element_miniscope/miniscope.py

```diff
@@ -0,0 +1,847 @@
+
+        if q_param:  # If the specified param-set already exists
+            pname = q_param.fetch1('paramset_id')
+            if pname == paramset_id:  # If the existed set has the same name: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
+                raise dj.DataJointError(
+                    'The specified param-set already exists - name: {}'.format(pname))
+        else:
+            cls.insert1(param_dict)
+
+@schema
+class ProcessingTask(dj.Manual):
+    definition = """
+    # Manual table marking a processing task to be triggered or manually processed
+    -> RecordingInfo
+    -> ProcessingParamSet
+    ---
+    processing_output_dir : varchar(255)    # relative to the root data directory
+    task_mode='load'      : enum('load', 'trigger') # 'load': load existing results
+                                                    # 'trigger': trigger procedure
+    """
+
+@schema
+class Processing(dj.Computed):
+    definition = """
+    -> ProcessingTask
+    ---
+    processing_time     : datetime  # time of generation of this set of processed, segmented results
+    package_version=''  : varchar(16)
+    """
+
+    def make(self, key):
+        task_mode = (ProcessingTask & key).fetch1('task_mode')
+        
+        output_dir = (ProcessingTask & key).fetch1('processing_output_dir')
+        output_dir = find_full_path(get_miniscope_root_data_dir(), output_dir)
+
+        if task_mode == 'load':
+            method, loaded_result = get_loader_result(key, ProcessingTask)
+            if method == 'caiman':
+                loaded_caiman = loaded_result
+                key = {**key, 'processing_time': loaded_caiman.creation_time}
+            else:
+                raise NotImplementedError(f'Loading of {method} data is not yet' 
+                                          f'supported')
+        elif task_mode == 'trigger':
+            method = (ProcessingTask * ProcessingParamSet * ProcessingMethod * \
+                        Recording & key).fetch1('processing_method')
+
+            if method == 'caiman':
+                import caiman
+                from element_interface.run_caiman import run_caiman
+
+                avi_files = (Recording * RecordingInfo * RecordingInfo.File \
+                                & key).fetch('recording_file_path')
+                avi_files = [find_full_path(get_miniscope_root_data_dir(), 
+                                    avi_file).as_posix() for avi_file in avi_files]
+
+                params = (ProcessingTask * ProcessingParamSet & key).fetch1('params')
+                sampling_rate = (ProcessingTask * Recording * RecordingInfo \
+                                    & key).fetch1('fps')
+
+                filename_hash = '.' + str(dict_to_uuid(dict(**key, **params)))
+
+                if not os.path.isfile(output_dir / filename_hash):
+                    run_caiman(file_paths=avi_files, 
```

Comment:
I'm seeing an error here, related to caiman's `fit_file`:
```
TypeError: fit_file() got an unexpected keyword argument 'output_dir'
```

Reply:
Ah yes, this feature is currently being evaluated in this [pull request](https://github.com/flatironinstitute/CaImAn/pull/948).

Reply:
Assuming a release this week, how should we document the current functionality to a potential user who sees the tutorial video next week?